### ISSUE TITLE
OpenSslCachedSystemStoreProvider refreshes certificates hard coded every 5 minutes which delays TLS connection handshake

### ISSUE THREAD

**Issue by AloisKraus** [dotnet/runtime#106394]
### Description

While debugging a performance issue of 120ms of delays in a Linux based web service we have found that during SSL handshake every 5 minutes over 100ms are spent in OpenSssCachedSystemProvider.LoadMachineStores. 

|| |+ System.Security.Cryptography!OpenSslCachedSystemStoreProvider.LoadMachineStores
|| | + System.Security.Cryptography!OpenSslCachedSystemStoreProvider.GetCollections
|| |  + System.Security.Cryptography!OpenSslCachedSystemStoreProvider.GetNativeCollections
|| |   + System.Security.Cryptography!OpenSslX509ChainProcessor.InitiateChain
|| |    + System.Security.Cryptography!ChainPal.BuildChainCore
|| |     + System.Security.Cryptography!ChainPal.BuildChain
|| |      + System.Security.Cryptography!X509Chain.Build
|| |       + System.Net.Security!CertificateValidation.BuildChainAndVerifyProperties
|| |        + System.Net.Security!SslStream.VerifyRemoteCertificate
|| |         + System.Net.Security!SslStream.CompleteHandshake
|| |          + System.Net.Security!System.Net.Security.SslStream+<ForceAuthenticationAsync>d__150`1[System.Net.Security.AsyncReadWriteAdapter].MoveNext()
|| |           + System.Private.CoreLib!ExecutionContext.RunInternal

This refresh time comes in hard coded in 
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCachedSystemStoreProvider.cs

```
    internal sealed class OpenSslCachedSystemStoreProvider : IStorePal
    {
        // These intervals are mostly arbitrary.
        // Prior to this refreshing cache the system collections were read just once per process, on the
        // assumption that system trust changes would happen before the process start (or would come
        // followed by a reboot for a kernel update, etc).
        // Customers requested something more often than "never" and 5 minutes seems like a reasonable
        // balance.
        private static readonly TimeSpan s_lastWriteRecheckInterval = TimeSpan.FromSeconds(5);
        private static readonly TimeSpan s_assumeInvalidInterval = TimeSpan.FromMinutes(5);

```

Can this hard code value be made configurable? 

### Regression?

This 5 minute timer was added as part of https://github.com/dotnet/runtime/commit/b20479c84943b3070f75bb971e059180dd8f4925 5 years ago.

**Comment by AloisKraus** [comment]
@vcsjones: Making that update async would be perfectly fine.

**Comment by bobbanks** [comment]
Would this issue occur when using ASP.NET Core Certificate Authentication with X509ChainTrustMode.CustomRootTrust?  I'm seeing similar symptoms (delays every 5 minutes) when running in Ubuntu WSL but not in Windows.

**Comment by bobbanks** [comment]
@bartonjs Am I correct in that this is not planned for 10.0?

**Comment by bartonjs** [comment]
@bobbanks Correct.  Almost all of the cryptography focus for 10 was in PQC, we just didn't have the time.

I wouldn't expect it to apply to CustomRootTrust, but maybe it runs through a codepath I wouldn't expect.

**Comment by bobbanks** [comment]
@bartonjs Appears OpenSslX509ChainProcessor.InitiateChain calls GetNativeCollections regardless of `trustMode`.  Do you concur?  I'm looking for any kind of workaround as I'm running into this on a low-powered device and it results in about 1.5s delay every 5 minutes.
